client/src/providers/UserProvider.tsx

```diff
@@ -2,14 +2,50 @@ import * as React from 'react';
 
 import type { User } from '../types';
 
-type UserContextT = [any | null, (user: any) => void];
+type UserContextT = [any | null, (user: User | null) => void];
 
-export const UserContext = React.createContext<UserContextT>([null, (user: any) => {}]);
+export const UserContext = React.createContext<UserContextT>([null, (user: User | null) => {}]);
 
 export function UserProvider(props: React.PropsWithChildren<{}>): JSX.Element {
   const { children } = props;
 
   const [user, setUser] = React.useState<User | null>(null);
 
+  async function fetchUser(): Promise<User | null> {
+    const res = await fetch('http://localhost:3001/api/auth/session', {
+      credentials: 'include',
+      headers: { 'Content-Type': 'application/json' },
+      method: 'POST',
+    });
+    const response = await res.json();
+    if (res.ok) {
+      return response.user;
+    }
+
+    return null;
+  }
+
+  function setUserTimeout(isInitialCall: boolean): void {
+    setTimeout(async () => {
+      if (!isInitialCall) {
+        const newUser = await fetchUser();
+        setUser(newUser);
+      }
+      setUserTimeout(false);
+    }, 59 * 60 * 1000);
+  }
```

Comment:
the value of the user object wouldn't actually change, so we wouldn't see any rerenders down the line
but idk if we need to be `setUser`ing...